lib/extendScalar.ml

```diff
@@ -58,6 +61,8 @@ module
     let of_int64 = to_narrow Narrow.of_int64
     and to_int64 = choose Narrow.to_int64 Wide.to_int64
 
+    let as_bool = choose Narrow.as_bool Wide.as_bool
+
```

Comment:
Hi again, here is the missing part. I do not think there are things such as wide booleans. See the definitions of `s_true` and `s_false` above. If so, we can write :
```suggestion
    l
    let as_bool = function
      | Narrow i -> Narrow.as_bool i
      | Wide i ->
          Warn.fatal "as_bool on wide scalar '%s'" @@ Wide.pp i
```